.github/workflows/upload.yml

```diff
@@ -0,0 +1,37 @@
+name: Upload
+on:
+  push:
+    branches:
+      - master
+jobs:
+  upload-build:
+    if: github.event.pull_request.merged == true
+    name: Upload build artifacts
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [12.x]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Restore node_modules
+        id: yarn-cache
+        uses: actions/cache@v2
+        with:
+          path: "**/node_modules"
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install
+        if: steps.yarn-cache.outputs.cache-hit != 'true'
+        run: CYPRESS_INSTALL_BINARY=0 yarn install
+      - name: Build
+        run: yarn build-all
```

Comment:
This will run `clean-all` which negates the benefit of doing the cache restore above. We should either change build-all to not clean or add a `build-ci` that doesn't run `clean-all`. We should also update the other CI workflow while we're at it: https://github.com/canonical-web-and-design/maas-ui/blob/master/.github/workflows/test.yml#L77